.github/workflows/chromatic.yml

```diff
@@ -29,8 +44,10 @@ jobs:
           filters: .github/file-paths.yaml
 
   fe-chromatic:
-    needs: files-changed
-    if: contains(github.event.pull_request.labels.*.name, 'chromatic') || (github.ref_name == 'master' && needs.files-changed.outputs.frontend_all == 'true')
+    needs: [files-changed, external-filter]
+    if: |
+      (needs.external-filter.result == 'success' && contains(github.event.pull_request.labels.*.name, 'chromatic')) ||
```

Comment:
in which cases external-filter will not finish with "success"?

Reply:
In case it's skipped.
And it will be skipped for external contributors on `pull_request` trigger.

```yml
if: |
  # Is it an internal PR? if yes -> proceed with the `pull_request` trigger 
  github.event.pull_request.head.repo.full_name == github.repository ||
  # If it is a fork it requires a `workflow-filter.yml` to run first. Otherwise, this is false and the job is skipped.
  # Once approved, Chromatic will be triggered the second time by a `workflow_run`
  github.event.workflow_run.conclusion == 'success'
```

For more info: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

Reply:
@uladzimirdev to illustrate this with an example for E2E tests (I tested this in my fork).

This is what happens once we approve 3rd party PR.
![image](https://github.com/metabase/metabase/assets/31325167/d9d2fe69-ffb2-418e-a8d3-f446dd64de87)

The first trigger is `pull_request` - it's skipped.
The second trigger says _completed_, which means that some other workflow completed and has triggered E2E tests (hence, on `workflow_run`.

The full mechanism is explained in this PR:
https://github.com/metabase/metabase/pull/34951